app/controllers/reports_controller.rb

```diff
@@ -5,6 +5,10 @@ def index
     @filters = EntryFilter.new(params[:entry_filter])
     @entries = entries
 
+    @total_hours = @entries.sum(:hours)
+    @total_billed= @entries.where(billed: true).sum :hours
+    @total_unbilled = @entries.where(billed: false).sum :hours
+
```

Comment:
So we're passing a lot of different variables to our view here, it's usually a sign that we need an object to hold them.

All of these have to do with the entries, so maybe we could have an `EntriesCollection` object or something? It would look something like this:

```ruby
class EntriesCollection
  attr_reader :entries
  def initialize(entries)
    @entries = entries
  end

  def total_hours
    entries.sum(:hours)
  end

  def total_biller
    entries.where(billed: true).sum :hours
  end

  def total_unbilled
    entries.where(billed: false).sum :hours
  end
end
```

Then in our controller:

```
def index
  # ...
  @entries_collection = EntriesCollection.new(entries)
end
```

And in our view:

```haml
%td= @entries_collection.total_hours
```